src/pint/models/timing_model.py

```diff
@@ -2639,6 +2697,269 @@ def __init__(self):
+            par = getattr(tm, tp)
+            for ap in par.aliases:
+                p2c_map[ap].append("timing_model")
+        return p2c_map
+
+    @lazyproperty
+    def _param_alias_map(self):
+        """Return the aliases map of all parameters
+
+        The returned map includes: 1. alias to PINT parameter name. 2. PINT
+        parameter name to pint parameter name. 3.prefix to PINT parameter name.
+
+        Notes
+        -----
+        Please use `alias_to_pint_param` method to map an alias to a PINT parameter.
+        """
+        alias = {}
+        for k, cp in self.components.items():
+            for p in cp.params:
+                par = getattr(cp, p)
+                # Check if an existing record
+                alias = self._add_alias_to_map(p, p, alias)
+                for als in par.aliases:
+                    alias = self._add_alias_to_map(als, p, alias)
+        tm = TimingModel()
+        for tp in tm.params:
+            par = getattr(tm, tp)
+            alias = self._add_alias_to_map(tp, tp, alias)
+            alias[tp] = tp
+            for als in par.aliases:
+                alias = self._add_alias_to_map(als, tp, alias)
+        return alias
+
+    @lazyproperty
+    def repeatable_param(self):
+        """Return the repeatable parameter map.
+        """
+        repeatable = []
+        for k, cp in self.components.items():
+            for p in cp.params:
+                par = getattr(cp, p)
+                if par.repeatable:
+                    repeatable.append(p)
+                    repeatable.append(par._parfile_name)
+                    # also add the aliases to the repeatable param
+                    for als in par.aliases:
+                        repeatable.append(als)
+        return list(set(repeatable))
+
+    @lazyproperty
+    def category_component_map(self):
+        """Return the mapping from category to component.
+        """
+        category = defaultdict(list)
+        for k, cp in self.components.items():
+            cat = cp.category
+            category[cat].append(k)
+        return category
+
+    @lazyproperty
+    def component_category_map(self):
+        """Return the mapping from component to category.
+        """
+        cp_ca = {}
+        for k, cp in self.components.items():
+            cp_ca[k] = cp.category
+        return cp_ca
+
+    @lazyproperty
+    def component_unique_params(self):
+        """Return the unique parameter names in each componens.
+
+        Note
+        ----
+        This function only returns the pint defined parameter name, not
+        including the aliases.
+        """
+        component_special_params = defaultdict(list)
+        for param, cps in self.param_component_map.items():
+            if len(cps) == 1:
+                component_special_params[cps[0]].append(param)
+        return component_special_params
+
+    def _add_alias_to_map(self, alias, param_name, alias_map):
```

Comment:
It is made for packing the same code. If you look at the code of the `_param_alias_map` this function is called again and again.

Reply:
Shouldn't the map be a property (maybe lazy) that is computed on the fly instead of requiring users to manually add aliases one at a time?

Reply:
This is a private function not meant for the users. The map is a lazy property. But the property call this function internally.

Reply:
Does it answer your question? Or what is your suggestion?

Reply:
I guess I don't understand why you're requiring other code to add aliases instead of, when you need to know what aliases exist, just going and querying the Parameters. Requiring the aliases to be added makes this a stateful process and leaves one to wonder what happens if someone forgets to add their alias, or how aliases are to be removed.

Reply:
I changed the name to '_check_alias_conflict'. Please take a look.